source/adapters/hip/enqueue.cpp

```diff
@@ -1239,27 +1240,20 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueMemBufferMap(
   UR_ASSERT(offset + size <= BufferImpl.getSize(),
             UR_RESULT_ERROR_INVALID_SIZE);
 
-  ur_result_t Result = UR_RESULT_ERROR_INVALID_OPERATION;
-  const bool IsPinned =
-      BufferImpl.MemAllocMode == BufferMem::AllocMode::AllocHostPtr;
-
-  // Currently no support for overlapping regions
-  if (BufferImpl.getMapPtr() != nullptr) {
-    return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+  auto MapPtr = BufferImpl.mapToPtr(size, offset, mapFlags);
+  if (!MapPtr) {
+    return UR_RESULT_ERROR_INVALID_MEM_OBJECT;
   }
 
-  // Allocate a pointer in the host to store the mapped information
-  auto HostPtr = BufferImpl.mapToPtr(size, offset, mapFlags);
-  *ppRetMap = std::get<BufferMem>(hBuffer->Mem).getMapPtr();
-  if (HostPtr) {
-    Result = UR_RESULT_SUCCESS;
-  }
+  const bool IsPinned =
+      BufferImpl.MemAllocMode == BufferMem::AllocMode::AllocHostPtr;
 
+  ur_result_t Result = UR_RESULT_SUCCESS;
```

Comment:
Nit: We sort of prefer using `UR_CHECK_ERROR(blah)` nowadays instead of `Result = blah`, so this var can usually be removed.

Reply:
(Unrelated to this patch but just a bit more info: if `blah` can return`UR_RESULT_ERROR_ADAPTER_SPECIFIC` we can't use `UR_CHECK_ERROR`, we need to return the return val like [here](https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/cuda/enqueue.cpp#L464))

Reply:
Thanks - I've updated to use the newer error checking style.